spec/plugins_spec.rb

```diff
@@ -13,10 +15,74 @@
 
   it "handles a message" do
     plugin = Linkbot::Plugin.plugins['mock'][:ptr]
-    response = Linkbot::Message.handle(Message.new("text", 1, "user", nil, :message, {}))
+    response = Linkbot::Message.handle(message)
 
     expect(plugin.messages.length).to eq 1
     expect(plugin.messages[0].body).to eq "text"
     expect(response).to eq ['text']
   end
+
+  context "has permission" do
+
+    it "when there is no room associated with a message" do
+      message.options = {}
+
+      expect(Linkbot::Plugin.has_permission?(message)).to be true
+    end
+
+    it "when no explicit permissions are set" do
+      ::Linkbot::Config["permissions"] = nil
+
+      expect(Linkbot::Plugin.has_permission?(message)).to be true
+    end
+
+    it "when permissions are set, but not for this room" do
+      ::Linkbot::Config["permissions"] = {
+        "not_this_room" => { "whitelist" => ["not_this_plugin"] } }
+
+      expect(Linkbot::Plugin.has_permission?(message)).to be true
+    end
+
+    it "when this room whitelists this plugin" do
+      ::Linkbot::Config["permissions"] = {
+        "this_room" => { "whitelist" => ["Linkbot::Plugin"] } }
+
+      expect(Linkbot::Plugin.has_permission?(message)).to be true
+    end
+
+    it "when this room blacklists other plugins, but not this one" do
+      ::Linkbot::Config["permissions"] = {
+        "this_room" => { "blacklist" => ["some", "other", "plugins"] } }
+
+      expect(Linkbot::Plugin.has_permission?(message)).to be true
+    end
+
+    it "when this room both whitelists and blacklists this plugin" do
+      ::Linkbot::Config["permissions"] = {
+        "this_room" => { "whitelist" => ["Linkbot::Plugin"],
+                         "blacklist" => ["Linkbot::Plugin"] } }
+      expect(Linkbot::Plugin.has_permission?(message)).to be true
+    end
+
+  end
+
+  context "does not have permission" do
+
+    it "when this room whitelists other plugins, but not this one" do
+      ::Linkbot::Config["permissions"] = {
+        "this_room" => { "whitelist" => ["some", "other", "plugins"] } }
+
+      expect(Linkbot::Plugin.has_permission?(message)).to be false
+    end
```

Comment:
Something to note: if a whitelist is set, a plugin must be included for it to have permission. Whitelist is default deny. So, I'm not sure [the comment at the bottom of the method](https://github.com/markolson/linkbot/blob/mark-refactor/lib/linkbot/plugin.rb#L53) is accurate.